File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.kt
class Solution {
    fun solution(s: String, skip: String, index: Int): String {
        var answer: String = ""
        var ascii_a = 'a'.toInt();
        var dics = List(26) {i -> (ascii_a+i).toChar() };
        dics = dics.filterNot { skip.contains(it) };
        
        for(ch in s) {
            var idx = dics.indexOf(ch);
            idx = (idx + index) % dics.size;
            answer += dics[idx];
        }
        return answer
    }
}

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.js
function solution(s, skip, index) {
    var answer = '';
    let ascii_a = 'a'.charCodeAt(0);
    let ascii_z = 'z'.charCodeAt(0);
    let dics = [];
    const skips = skip.split('');
    
    for(let i = ascii_a; i <= ascii_z; i++) {
        dics.push(String.fromCharCode(i));
    }
    
    dics = dics.filter(item => !skips.includes(item));
    
    for(let c of s) {
        let idx = dics.indexOf(c);
        idx = (idx + index)%dics.length;
        answer += dics[idx];
    }
    

    return answer;
}